### PR TITLE
smart turret implementation

### PR DESCRIPTION
**Pull request by 0xxlegolas**
smart turret implementation
- added functions to support smart turret
- unit tests 
- added corpId to smart character

### CODE REVIEW

**Review by ledablack — APPROVED** [review]

File: mud-contracts/world/src/modules/smart-turret/SmartTurretLib.sol [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+  ) internal returns (Target[] memory returnTargetQueue) {
+    bytes memory data = world.iface.call(
+      world.namespace.smartTurretSystemId(),
+      abi.encodeCall(
+        ISmartTurret.aggression,
+        (smartTurretId, aggressorCharacterId, aggressorHp, victimItemId, victimHp, priorityQueue, chargesLeft)
+      )
+    );
+    return abi.decode(data, (Target[]));
+  }
```
Comment (vayan) [inline mud-contracts/world/src/modules/smart-turret/SmartTurretLib.sol]: to normalize naming 🙏 (should be done throughout the changes)

```suggestion
  function inProximity(
    World memory world,
    uint256 smartTurretId,
    uint256 characterId,
    Target[] memory targetQueue,
    uint256 chargesLeft,
    uint256 hpRatio
  ) internal returns (Target[] memory returnTargetQueue) {
    bytes memory data = world.iface.call(
      world.namespace.smartTurretSystemId(),
      abi.encodeCall(ISmartTurret.inProximity, (smartTurretId, characterId, targetQueue, remainingAmmo, hpRatio))
    );
    return abi.decode(data, (Target[]));
  }

  function aggression(
    World memory world,
    uint256 smartTurretId,
    uint256 aggressorCharacterId,
    uint256 aggressorHpRatio,
    uint256 victimItemId,
    uint256 victimHpRatio,
    Target[] memory priorityQueue,
    uint256 chargesLeft
  ) internal returns (Target[] memory returnTargetQueue) {
    bytes memory data = world.iface.call(
      world.namespace.smartTurretSystemId(),
      abi.encodeCall(
        ISmartTurret.aggression,
        (smartTurretId, aggressorCharacterId, aggressorHp, victimItemId, victimHp, priorityQueue, chargesLeft)
      )
    );
    return abi.decode(data, (Target[]));
  }
```

**Comment by 0xxlegolas** [comment]
@vayan  Could you check now ? after the naming changes ?

**Comment by vayan** [comment]
All good! Thanks a lot 💙 


:shipit:

**Review by vayan — APPROVED** [review]

**Review by fc-tezza — APPROVED** [review]
LGTM

**Pull request merged**